### ISSUE TITLE
Events not being trigged on some zips

### ISSUE THREAD

**Issue by sadikyalcin** [ZJONSSON/node-unzipper#78]
Close, error or finish events  are not triggering on certain zips and I cannot seem to find out why. I had this issue on `unzip` too - and this only happens on Windows.

    this.extract = function(id, onSuccess, onError){
        var zip = app.path.join(nw.App.dataPath, 'ca-user-data/' + app.session.domain + '/' + app.user.name + '/presentations/' + id + '.zip');
        var target = app.path.join(nw.App.dataPath, 'ca-user-data/' + app.session.domain + '/' + app.user.name + '/presentations/' + id);
        var zipper = app.fs.createReadStream(zip).pipe(app.unzip.Extract({ path: target }));
        
        zipper.on('close', function(e){
            onSuccess();
        });

        zipper.on('error', function(e){
            console.log('Error: ', e);
            onError(e);
        });

        zipper.on('drain', function(e){
            console.log('Drained: ', e);
        });

        zipper.on('entry', function(e){
            console.log('Entry: ', e);
        });

        zipper.on('finish', function(e){
            console.log('Finished: ', e);
        });
    };

**Comment by ZJONSSON** [comment]
The unzipper uses a pullstream to extract data.   The Pullstream stops when you are not pulling from it.  So in the code above, could you try to add autodrain to the entry event (which ensures all data is pulled from the entry, so we can get the next entry):

```js
zipper.on('entry', function(e) {
  console.log('Entry: ', e);
  e.autodrain();
});
```

**Comment by sadikyalcin** [comment]
I will do that but I do have extra info (After days and days of digging!). I've noticed that there is an extra `\` in some directories within the zip. Apparently, which Windows fails to extract and OSX does - as seen by this [comment](http://php.net/manual/en/book.zip.php) left on the PHP manual 9 years ago and replicated it on my end. What a nightmare!

Is there anything I can do to handle this?

**Comment by ZJONSSON** [comment]
If you are extracting manually then you can obviously handle this yourself directly.   If however you are using the `.extract` method then we might want to replace any consecutive slashes with a single slash.  Do you want to make a quick PR with a test case?    If not I can throw on in when I have time.

**Comment by ZJONSSON** [comment]
Should be fixed by #121   Please reopen if issue persists
